src/java.base/share/classes/java/lang/runtime/ObjectMethods.java

```diff
@@ -324,6 +324,7 @@ private static MethodHandle makeToString(Class<?> receiverClass,
      *                     if invoked by a condy
      * @throws IllegalArgumentException if the bootstrap arguments are invalid
      *                                  or inconsistent
+     * @throws NullPointerException if any argument but {@code lookup} is {@code null}
```

Comment:
`names` may be null if the {@code methodName} is {@code "equals"} or {@code "hashCode"}.    This should be captured here.

Reply:
Hi Mandy, I have changed the implementation of the method to explicitly require all arguments but lookup to be non-null as suggested by Brian. I have also covered, I think, all the missing test cases in test `ObjectMethodsTest`, thanks for your comments.

Reply:
I think you meant requiring `names` to be non-null but `lookup` may still be null.  It's okay to change the spec to require `names` to be non-null.  

Probably better to mention in `@param names` that `names` is ignored when the `methodName` is `equals` or `hashCode`.

Reply:
no I think I prefer to force names to be non-null all the time, that way we offer a more consistent interface to users. They don't have to remember what case was for which names can be null.